File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    su=0
    s = set()
    lenght = len(elements)
    for n in range(1,lenght+1):
        elements.append(elements[n-1])
        for i in range(len(elements)-n):
            su = sum(elements[i:i+n])
            s.add(su)
    return len(s)
    return answer

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>
#include <algorithm>
#include <numeric>

using namespace std;

int solution(vector<int> elements) {
    int answer = 0;
    set<int> s;
    for(int i = 1; i <= elements.size(); i++){
        int sum = 0;
        for(int j = 0; j < elements.size(); j++){
            if(j+i >= elements.size()){
                sum = accumulate(elements.begin()+j, elements.end(), 0);
                sum += accumulate(elements.begin(), elements.begin() + (i+j)-elements.size(), 0);
            }
            else{
                sum = accumulate(elements.begin()+j, elements.begin()+i+j, 0);     
            }
            s.insert(sum);
        }
    }
    return s.size();
}